neo/Consensus/RecoveryMessage.cs

```diff
@@ -1,20 +1,20 @@
-﻿using System.IO;
-using Neo.IO;
+﻿using Neo.IO;
+using Neo.Ledger;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
 
 namespace Neo.Consensus
 {
-    internal class RecoveryMessage : ConsensusMessage
+    internal partial class RecoveryMessage : ConsensusMessage
     {
-        public byte[][] ChangeViewWitnessInvocationScripts;
-        public uint[] ChangeViewTimestamps;
-        public byte[] OriginalChangeViewNumbers;
+        public Dictionary<ushort, ChangeViewPayloadCompact> ChangeViewMessages;
```

Comment:
If you are going stay with a Dictionary instead of an array, there is one optimization that could be done here. Using the key of ushort may seem better because the `MaxValidators` is 1024, which fits in a `ushort`; however, just using `int` is likely optimized better by the compiler. Again, another thing that likely wouldn't make a big difference when profiling. You could change this to `int` though and still serialize and deserialize it as a `short` for saving space in the payload.

Reply:
Done.